build/v1/types.go

```diff
@@ -1335,3 +1348,59 @@ type SecretSpec struct {
 	// mountPath is the path at which to mount the secret
 	MountPath string `json:"mountPath" protobuf:"bytes,2,opt,name=mountPath"`
 }
+
+// BuildVolume describes a volume that is made available to build pods,
+// such that it can be mounted into buildah's runtime environment.
+// Only a subset of Kubernetes Volume sources are supported.
+type BuildVolume struct {
+	// name is a unique identifier for this BuildVolume.
+	// It must conform to the Kubernetes DNS label standard and be unique within the pod.
+	// It cannot collide with volumes that are always added by the build controller.
```

Comment:
can you specify what happens if it does collide, here?  Users of the api will not have any idea what volumes the build controller adds in order to avoid them, so even better might be providing some indication of the naming scheme the build controller uses so users can avoid colliding.

Reply:
We actually take the `name` that they provide and append `-user-build-volume` to it, so there is probably a very very low chance of collision. I also don't believe that the build controller has a naming scheme that it uses for other types of mounts. They just are what they are. I can provide an example once my cluster finished launching

Reply:
it does have a naming scheme, though there may be several for different types of things being mounted.  But for example for secrets i think it uses a pattern like "Sourcesecret_1", "sourcesecret_2" or some such thing.

Reply:
anyway the main clarification i want to see is rather than saying "it cannot collide" it should either say:
1) names that collide will result in a validation error
or
2) whatever the behavior is if a collision occurs

Reply:
updated in next push